src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,218 @@
+package duke;
+
+
+import duke.commands.Command;
+import duke.exceptions.DukeException;
+
+import java.io.InputStream;
+import java.util.Scanner;
+
+
+/**
+ * Main class for the Duke programme.
+ */
+public class Duke {
+
+    private final DukeList taskList = new DukeList();
+    private final Ui ui;
+
+
+    /**
+     * Constructor for Duke.
+     */
+    public Duke() {
+        this.ui = new Ui();
+    }
+
+
+    /**
+     * Constructor for Duke.
+     *
+     * @param inputStream input stream to read user inputs from.
+     */
+    public Duke(InputStream inputStream) {
+        this.ui = new Ui(inputStream);
+    }
+
+
+    /**
+     * Determine if programme should quit given the input message.
+     *
+     * @param msgInput Message input from user.
+     * @return if programme should quit.
+     */
+    private static Boolean shouldQuit(String msgInput) {
+        final String QUIT_STRING = "bye";
+
+        return msgInput.equals(QUIT_STRING);
+    }
+
+
+    /**
+     * Marks an item as done.
+     *
+     * @param inputTextArr String array of the input text, split by " ".
+     * @return statusMSg to be printed.
+     * @throws IndexOutOfBoundsException Invalid index given or input text array is invalid.
+     */
+    private String markAsDone(String[] inputTextArr) throws IndexOutOfBoundsException {
+        try {
+            int index = Integer.parseInt(inputTextArr[1]);
+            String statusMsg = this.taskList.markAsDone(index);
+            return statusMsg;
+
+        } catch (ArrayIndexOutOfBoundsException e) {
+            // forwards ArrayIndexOutOfBoundsException from Integer.parseInt()
+            // inputTextArr is invalid.
+            throw new ArrayIndexOutOfBoundsException("Invalid inputTextArr.");
+        } catch (IndexOutOfBoundsException e) {
+            // exception thrown from DukeList.markAsDone()
+            // index given in input text is invalid.
+            throw new IndexOutOfBoundsException("Invalid index given.");
+        }
+    }
+
+
+    /**
+     * Deletes an item from the taskList.
+     *
+     * @param inputTextArr String array of the input text, split by " "
+     * @return statusMsg to be printed.
+     * @throws IndexOutOfBoundsException Invalid index given or input text array is invalid.
+     */
+    private String delete(String[] inputTextArr) throws IndexOutOfBoundsException {
+        try {
+            int index = Integer.parseInt(inputTextArr[1]);
+            String statusMsg = this.taskList.delete(index);
+            return statusMsg;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            // forwards ArrayIndexOutOfBoundsException from Integer.parseInt()
+            // inputTextArr is invalid.
+            throw new ArrayIndexOutOfBoundsException("Invalid inputTextArr.");
```

Comment:
Perhaps you can throw a child of DukeException here to tell the user that they forgot to input the number of the task to be deleted for better UI.